extension/json/src/functions/table_functions/json_scan.cpp

```diff
@@ -380,11 +482,15 @@ bool JSONScanLocalState::readNextBuffer() {
 
     if (currentReader && currentBufferHandle) {
         if (--currentBufferHandle->numReaders == 0) {
```

Comment:
@acquamarin is this operation thread-safe or should `numReaders` (and the `lineOrObjectCountInBuffer` I added) be atomic?

Reply:
Yes, the numer of readers should be defined as `std::atomic<uint64_t>`

Reply:
updated